docs/tutorial/gatsby-image-tutorial/gatsby-image-tutorial.md

```diff
@@ -0,0 +1,361 @@
+
+The query will return a processed image at the highest depth. That is, the deepest part of the return object. The returned result will be automatically passed into the component and attached to the `data` prop. You can then display the image using JSX to automatically output responsive, highly performant HTML.
+
+To display the image, start by importing the component provided by `gatsby-image`.
+
+```jsx
+import Img from "gatsby-image"
+```
+
+Now you can use it. Note that the key for pointing to the image corresponds to the way in which the image was processed. In this example that is `fixed`.
+
+```jsx
+<Img
+  className="headshot"
+  fixed={data.file.childImageSharp.fixed}
+  alt="headshot"
+/>
+```
+
+Here is the query and usage all put together:
+
+```jsx:title=src/pages/index.js
+import React from "react"
+import { graphql } from "gatsby"
+import Img from "gatsby-image"
+import Layout from "../components/layout"
+
+const HomePage = ({ data }) => {
+  return (
+    <Layout>
+      <Img
+        className="headshot"
+        fixed={data.file.childImageSharp.fixed}
+        alt="headshot"
+      />
+    </Layout>
+  )
+}
+
+export const query = graphql`
+  query {
+    file(relativePath: { eq: "headers/headshot.jpg" }) {
+      childImageSharp {
+        fixed(width: 125, height: 125) {
+          ...GatsbyImageSharpFixed
+        }
+      }
+    }
+  }
+`
+export default HomePage
+```
+
+## Querying for multiple images from YAML data
+
+Another way to source images is through YAML (or Markdown). This example uses the `gatsby-transformer-yaml` plugin to query the YAML files. More information about that plugin can be found in the [Gatsby plugin library](https://www.gatsbyjs.org/packages/gatsby-transformer-yaml/?=gatsby-transformer-yaml).
+
+Here's an example of a query from a list of conferences in a YAML file:
+
+```graphql
+{
+  allSpeakingYaml {
+    edges {
+      node {
+        conference
+        year
+      }
+    }
+  }
+}
+```
+
+In this case the query starts with `allSpeakingYaml` to direct `graphql` to look for this data in the `speaking.yaml` file in your `src/data` folder referenced in `gatsby-config.js`. If you want to query a file named `blog.yaml`, for example, you'd start the query with `allBlogYaml`.
+
+### Troubleshooting queries
+
+A common error you might run into is `Unknown field allSpeakingYaml on type Query`, where `allSpeakingYaml` is replaced with something specific to your code. What the error boils down to is that `graphql` is unable to find a matching result. This can mean it couldn't find a file that matched the first line, meaning that no `speaking.yaml` file exists.
```

Comment:
If we can cover the query gotchas in a few bullet points and keep the info limited to images, it will make this more succinct. See comments above about moving the image query stuff up to cover it sooner!